signer/src/storage/model.rs

```diff
@@ -10,11 +10,172 @@ use serde::Deserialize;
+                    })
+                    .collect();
+
+                SweepTransaction {
+                    id: None,
+                    txid: transaction.tx.compute_txid().into(),
+                    utxo_txid: transaction.signer_utxo.utxo.outpoint.txid.into(),
+                    utxo_output_index: transaction.signer_utxo.utxo.outpoint.vout,
+                    amount: transaction.output_amounts(),
+                    fee: transaction.tx_fee,
+                    fee_rate: transaction.signer_utxo.fee_rate,
+                    is_broadcast: false,
+                    swept_deposits,
+                    swept_withdrawals,
+                }
+            })
+            .collect();
+
+        SweepTransactionPackage {
+            created_at_block_hash: chain_tip.into(),
+            market_fee_rate,
+            transactions,
+        }
+    }
+}
+
+/// Represents a single transaction which is part of a sweep transaction package
+/// which has been broadcast to the Bitcoin network.
+#[derive(Debug, Clone, PartialEq, PartialOrd, sqlx::FromRow)]
+#[cfg_attr(feature = "testing", derive(fake::Dummy))]
+pub struct SweepTransaction {
+    /// The internal id of the packaged transaction.
+    #[sqlx(try_from = "Option<i64>")]
+    #[cfg_attr(feature = "testing", dummy(default))]
+    pub id: Option<i64>,
+    /// The Bitcoin transaction id.
+    pub txid: BitcoinTxId,
+    /// The transaction id of the signer UTXO consumed by this transaction.
+    pub utxo_txid: BitcoinTxId,
+    /// The index of the signer UTXO consumed by this transaction.
+    #[sqlx(try_from = "i32")]
+    pub utxo_output_index: u32,
+    /// The total amount of this transaction.
+    #[sqlx(try_from = "i64")]
+    #[cfg_attr(feature = "testing", dummy(faker = "1_000_000..1_000_000_000"))]
+    pub amount: u64,
+    /// The fee paid for this transaction.
+    #[sqlx(try_from = "i64")]
+    #[cfg_attr(feature = "testing", dummy(faker = "1_000_000..1_000_000_000"))]
+    pub fee: u64,
+    /// The fee rate of this transaction in sats/vByte.
+    #[cfg_attr(feature = "testing", dummy(default))]
+    pub fee_rate: f64,
+    /// Whether or not this transaction has been successfully broadcast to the
+    /// Bitcoin network.
+    #[cfg_attr(feature = "testing", dummy(default))]
+    pub is_broadcast: bool,
+
+    /// List of deposits which were swept-in by this transaction.
+    #[sqlx(skip)]
+    pub swept_deposits: Vec<SweptDeposit>,
+    /// List of withdrawals which were swept-out by this transaction.
+    #[sqlx(skip)]
+    pub swept_withdrawals: Vec<SweptWithdrawal>,
+}
+
+/// Represents a single deposit which has been swept-in by a sweep transaction
+/// package.
+#[derive(Debug, Clone, Hash, PartialEq, Eq, PartialOrd, Ord, sqlx::FromRow)]
+#[cfg_attr(feature = "testing", derive(fake::Dummy))]
+pub struct SweptDeposit {
+    /// The index of the deposit output in the sBTC sweep transaction.
+    #[sqlx(try_from = "i32")]
+    pub output_index: u32,
```

Comment:
I don't think that we need to add this. It's slower but if we need to know this we should just scan the inputs by the outpoint (the transaction ID and the output index).

Reply:
Updated in [81933a7](https://github.com/stacks-network/sbtc/pull/710/commits/81933a7404312197b4e1d4795e498a7d81925929)